.github/workflows/publish-major-minor.yml

```diff
@@ -0,0 +1,33 @@
+name: "Publish Major-Minor-Tags"
+on:
+  push:
```

Comment:
Honestly, I'm lately preferring the `create` event because it has closer semantics to what we're attempting to trigger. But using it requires a post-condition.

WDYT?

Reply:
I mostly use `push` due to sheer habit, there were a lot fewer triggers (at least in the docs) when I started adopting gh-actions as soon as they came out of beta.
The [docs](https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#create) are kinda unclear on how to use it, would simply changing `push` for `create` work? Or what is the proper pattern, only to trigger on tags that start with `v`

I guess semantically the closes would be to use [`release`](https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#release).
```yaml
on:
  release:
    types: [published]
```
I use this trigger for my actions and it works fine, but I'm not sure how this will play out with `dev`, `pre` and `post` tags, since those will most likely not published.
The other `release` type to use would be `created`, which wouldn't force one to release on the markedplace, but it could still happen by accident (I think to remember that the checkbox is checked by default after fist publishing).

Reply:
The usage of `create` is
It's
```yaml
on:
  create:

jobs:
  job-name:
    if: github.event.ref_type == 'tag'  # to filter out branch creation
```
(because it doesn't have "native" filters)

I usually create the tags using local Git and push them from CLI. And only after that I publish releases. So I'm not sure if `release` would be suitable. OTOH this workflow could auto-publish a release on tag creation. You already have a way of saying if the "pre-release" checkbox should be set.

Also, I'm fine with publishing pre-releases to Marketplace. I've done it before.
See https://github.com/pypa/gh-action-pypi-publish/releases/tag/v1.0.0a0 — it has a pre-release checkbox. And it's marketplace page https://github.com/marketplace/actions/pypi-publish?version=v1.0.0a0 also has a pre-release marker, just like PyPI.

Reply:
I think auto-publishing is a bit too much automation since the release messages often need some handcrafted work. e.g. markdown formatting, which can't be extracted from the tag.

As for the `published` event, I don't know how github determines what the current release is, my guess is that it sorts them by published date. If that assumption is correct a `post` release could mess up what version is displayed on the marketplace. E.g. you have a `v2.0.0` release and after that release `v1.4.1.post0` to trigger the workflow, `v1.4.1.post0` might be shown a current version. So IMHO `[created, edited]` would be closer to the desired behavior.
Also using `release` would make it more reasonable to add a link to the release to the created tags (see https://github.com/pypa/gh-action-pypi-publish/pull/45#discussion_r518826619).

As for the semantics of `create` vs. `push`, I think this boils down to different reference systems `remote` vs. `locally` (sorry physics student here 😆 ). 
I personally prefer the more concise syntax of `push`, where
```yaml
on:
  push:
    tags:
      - "v*"
```
gives you the same functionality as 
```yaml
on:
  create:

jobs:
  job-name:
    if: github.event.ref_type == 'tag' && startsWith( github.ref, 'v')
```

Reply:
I'm not sure about the markdown part. First of all, it's possible to have tag annotations with markdown baked-in but I wouldn't care about it too much: when you mark a release as published to the Marketplace, the new version appears there but the release description does not show up there. Instead, it just contains README content.
So it's okay to just auto-publish and then, if needed, I'd just correct it.
FWIW we shouldn't care about ordering the tags: it'll be like this regardless of whether a robot or a human makes the release... And we're not going to avoid putting `.post` tags on the marketplace while moving the major tag there in Git.